src/librustc_mir/interpret/memory.rs

```diff
@@ -74,11 +73,10 @@ impl<'a, 'mir, 'tcx, M> PartialEq for Memory<'a, 'mir, 'tcx, M>
     }
 }
 
-impl<'a, 'mir, 'tcx, M> Hash for Memory<'a, 'mir, 'tcx, M>
+impl<'a, 'b, 'mir, 'tcx, M> HashStable<StableHashingContext<'b>> for Memory<'a, 'mir, 'tcx, M>
```

Comment:
do we still need this impl now that we hash only the reachable memory?

Reply:
Good question, I guess not, though we still need to hash `Memory::data` and `Memory::cur_frame`, however that can be done explicitly by the impl `HashStable` for `EvalSnapshot`, meaning probably can get rid of this impl.